File: src/ts/Models/Entities/ArtistAlbum.ts
export default class ArtistAlbum {
    private _artistId: string;
    private _albumId: string;

    public get ArtistId(): string {
        return this._artistId;
    }
    public get AlbumId(): string {
        return this._albumId;
    }

    public constructor(artistId: string, albumId: string) {
        this._artistId = artistId;
        this._albumId = albumId;
    }
}

File: aspCore/Models/Relations/ArtistAlbum.cs
using MusicFront.Models.Albums;
using MusicFront.Models.Artists;
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MusicFront.Models.Relations
{
    [Table("artist_albums")]
    [JsonObject(MemberSerialization.OptIn)]
    public class ArtistAlbum
    {
        [Required]
        [JsonProperty]
        public int ArtistId { get; set; }

        [Required]
        [JsonProperty]
        public int AlbumId { get; set; }

        [ForeignKey("ArtistId")]
        public Artist Artist { get; set; }

        [ForeignKey("AlbumId")]
        public Album Album { get; set; }
    }
}
